### PR TITLE
Fix puzzle priority

### PR DESCRIPTION
**Pull request by marthinwurer**
We'll see if I implemented this right. The first puzzle ends up at the main phase.

### CODE REVIEW

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java [hunk 1]
```diff
@@ -264,20 +264,83 @@ private boolean isSkippingPhase(final PhaseType phase) {
         }
     }
 
+    public boolean givePriorityToPlayer() {
+        if (isSkippingPhase(phase)) {
+            return true;
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/phase/PhaseHandler.java:269]: why would this return true when it gets skipped? 😕

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java [hunk 2]
```diff
@@ -1281,7 +1311,7 @@ public String debugPrintState(boolean hasPriority) {
 
     // just to avoid exposing variable to outer classes
     public void onStackResolved() {
-        givePriorityToPlayer = true;
+//        givePriorityToPlayer = true;
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/phase/PhaseHandler.java:1314]: was it already obsolete or why this change?
Seems to satisfy 117.3b

**Comment by Agetian** [comment]
What's the exact issue that you are trying to solve?

**Comment by marthinwurer** [comment]
This one reported on the discord: https://discord.com/channels/267367946135928833/1156369694958293063

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java (hunk 1 above)
Comment (marthinwurer) [reply forge-game/src/main/java/forge/game/phase/PhaseHandler.java:269]: Good catch, it should be false.

**Comment by kevlahnota** [comment]
I tried this fix but it produces NPE when using Ninjutsu ability, to reproduce declare an attacker ie raging goblin, then on declare blockers step, activate ninjutsu ability targeting raging goblin then let the ability resolve, NPE is thrown.. 
```
java.lang.NullPointerException: null
    at forge.game.ability.effects.ChangeZoneEffect.changeKnownOriginResolve(ChangeZoneEffect.java:702)
    at forge.game.ability.effects.ChangeZoneEffect.resolve(ChangeZoneEffect.java:438)
    at forge.game.ability.AbilityApiBased.resolve(AbilityApiBased.java:41)
    at forge.game.ability.AbilityUtils.resolveApiAbility(AbilityUtils.java:1427)
    at forge.game.ability.AbilityUtils.resolve(AbilityUtils.java:1395)
    at forge.game.zone.MagicStack.resolveStack(MagicStack.java:547)
    at forge.game.phase.PhaseHandler.mainLoopStep(PhaseHandler.java:1257)
    at forge.game.phase.PhaseHandler.startFirstTurn(PhaseHandler.java:1142)
    at forge.game.GameAction.startGame(GameAction.java:2108)
    at forge.game.Match.startGame(Match.java:105)
    at forge.gamemodes.match.HostedMatch$2.run(HostedMatch.java:255)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)
```

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java (hunk 2 above)
Comment (kevlahnota) [reply forge-game/src/main/java/forge/game/phase/PhaseHandler.java:1317]: @Agetian It works fine here, Don't really know if this covers 117.3b  correctly. 
image

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java (hunk 1 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/phase/PhaseHandler.java:264]: why resolve, still seems wrong...?

File: forge-game/src/main/java/forge/game/phase/PhaseHandler.java (hunk 2 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/phase/PhaseHandler.java:1317]: Well my question was did it have to be commented out now to make this fix work at all?

**Comment by kevlahnota** [comment]
Karn Liberated fails restarting the game since the restartgameeffect relies on the old phasehandler fallback code.

**Comment by Agetian** [comment]
Is the original contributor still available to make the necessary followup changes to make this fix work without breaking other aspects of the game?
If it proves to be too difficult or too time-consuming to devise a working fix, I suggest that we either (a) roll back the original change that caused the Puzzle Mode to break, until a working solution is found *or* (b) implement a simpler "workaround" fix in devAdvancePhase (giving priority to the player in it) that unbreaks Puzzle Mode while keeping the originally contributed MR intact. This seems to work in my local testing, though I haven't tested all the possible corner cases. Whether we go for (b) or not, of course, also depends on whether anything else is broken (except Puzzle Mode) without a followup fix. 
What do you think?

**Comment by tool4ever** [comment]
Well seeing that @kevlahnota discovered another problem already existing after #3669 I'm leaning towards reverting that now before it gets too difficult (unless that's another easy fix)
